File: 4lesson.py
import telebot
import requests
import os
from datetime import datetime

# Инициализация бота
TOKEN = "your token"
bot = telebot.TeleBot(TOKEN)

# Функция для получения текущего времени
def get_time():
    return datetime.now().strftime("%Y-%m-%d %H:%M:%S")

# Функция для получения погоды через wttr.in (без API-ключей)
def get_weather(location):
    url = f"https://wttr.in/{location}?format=%C+%t&lang=ru"
    try:
        response = requests.get(url)
        if response.status_code != 200:
            return "Ошибка: не удалось получить данные о погоде."
        
        return f"Погода в {location}: {response.text}"
    except Exception as e:
        return f"Ошибка получения погоды: {str(e)}"


# Функция для обработки запросов к нейросети (заглушка)
def process_neural_request(text):
    return f"Нейросеть обработала запрос: {text}"

# Обработчик команды /start
@bot.message_handler(commands=['start'])
def send_welcome(message):
    bot.reply_to(message, "Привет! Я помощник-бот. Отправь /help, чтобы узнать мои команды.")

# Обработчик команды /help
@bot.message_handler(commands=['help'])
def send_help(message):
    help_text = "Я умею:\n"
    help_text += "/start - Запуск бота\n"
    help_text += "/help - Список команд\n"
    help_text += "/time - Узнать текущее время\n"
    help_text += "Введите город, чтобы узнать погоду (пример: Москва)\n"
    help_text += "Введите любой текст для обработки нейросетью"
    bot.reply_to(message, help_text)

# Обработчик команды /time
@bot.message_handler(commands=['time'])
def send_time(message):
    bot.reply_to(message, f"Текущее время: {get_time()}")

# Обработчик текстовых сообщений
@bot.message_handler(func=lambda message: True)
def handle_text(message):
    text = message.text.strip()
    if text.replace(" ", "").isalpha():  # Проверяем, что введен город (слово без цифр)
        bot.reply_to(message, get_weather(text))
    else:
        bot.reply_to(message, process_neural_request(text))

if __name__ == "__main__":
    bot.polling(none_stop=True)

File: 4lesson.go
package main

import (
	"fmt"
	"log"
	"net/http"
	//"os"
	"strings"
	"time"
	"unicode"
	"github.com/go-telegram-bot-api/telegram-bot-api/v5"
)

// Функция для получения текущего времени
func getTime() string {
	return time.Now().Format("2006-01-02 15:04:05")
}

// Функция для получения погоды через wttr.in (без API-ключей)
func getWeather(location string) string {
	url := fmt.Sprintf("https://wttr.in/%s?format=%%C+%%t&lang=ru", location)
	resp, err := http.Get(url)
	if err != nil {
		return "Ошибка: не удалось получить данные о погоде."
	}
	defer resp.Body.Close()

	var weather string
	_, err = fmt.Fscan(resp.Body, &weather)
	if err != nil {
		return "Ошибка при чтении ответа."
	}
	return fmt.Sprintf("Погода в %s: %s", location, weather)
}

// Функция-заглушка для обработки запросов к нейросети
func processNeuralRequest(text string) string {
	return fmt.Sprintf("Нейросеть обработала запрос: %s", text)
}

func main() {
	// Читаем токен бота из переменной окружения
	token := "token"
	if token == "" {
		log.Fatal("TELEGRAM_BOT_TOKEN не установлен")
	}

	bot, err := tgbotapi.NewBotAPI(token)
	if err != nil {
		log.Fatal(err)
	}

	bot.Debug = true
	log.Printf("Авторизован как %s", bot.Self.UserName)

	updateConfig := tgbotapi.NewUpdate(0)
	updateConfig.Timeout = 60
	updates := bot.GetUpdatesChan(updateConfig)

	for update := range updates {
		if update.Message == nil {
			continue
		}

		switch update.Message.Text {
		case "/start":
			msg := tgbotapi.NewMessage(update.Message.Chat.ID, "Привет! Я помощник-бот. Отправь /help, чтобы узнать мои команды.")
			bot.Send(msg)
		case "/help":
			helpText := "Я умею:\n"
			helpText += "/start - Запуск бота\n"
			helpText += "/help - Список команд\n"
			helpText += "/time - Узнать текущее время\n"
			helpText += "Введите город, чтобы узнать погоду (пример: Москва)\n"
			helpText += "Введите любой текст для обработки нейросетью"
			msg := tgbotapi.NewMessage(update.Message.Chat.ID, helpText)
			bot.Send(msg)
		case "/time":
			msg := tgbotapi.NewMessage(update.Message.Chat.ID, fmt.Sprintf("Текущее время: %s", getTime()))
			bot.Send(msg)
		default:
			text := strings.TrimSpace(update.Message.Text)
			if strings.IndexFunc(text, func(r rune) bool { return !(unicode.IsLetter(r) || unicode.IsSpace(r)) }) == -1 {

				msg := tgbotapi.NewMessage(update.Message.Chat.ID, getWeather(text))
				bot.Send(msg)
			} else {
				msg := tgbotapi.NewMessage(update.Message.Chat.ID, processNeuralRequest(text))
				bot.Send(msg)
			}
		}
	}
}
